core/directory.go

```diff
@@ -496,6 +498,68 @@ func (dir *Directory) WithNewFileDagOp(ctx context.Context, dest string, content
 	return dir, nil
 }
 
+func (dir *Directory) WithPatch(ctx context.Context, patch string) (*Directory, error) {
```

Comment:
I think this is pretty reasonable, but it does feel a bit weird that it's a git operation not associated with the `Git` type (see soooomewhat similar discussion of patches in the context of https://github.com/dagger/dagger/issues/7202#issuecomment-2966176223).

Something like `dir.AsGit().WithPatch().Tree()` feels like the thing I'd want there? But then there's not really a concept of a staging area there, so bleh, don't really know.

---

I'm happy to introduce this method now, and keep working on how this integrates with the Git API down the road (with the caveat we might have multiple ways of doing it in the future?).

Reply:
I don't think this is a git-specific operation. At least in the context of coding agents, it's pretty standard for regular file operations to include either a search-replace, or a patch operation. I don't know if that's the use case @vito had in mind, but it's definitely a valid one. And it's unrelated to git.

Reply:
Yeah I had to look this up - this is unrelated to Git, just happens to use `git apply` under the hood, for maximal compatibility with fun things like binary diffs. Despite using the `git` CLI it doesn't need to even run against a Git repo - any directory will do.